### ISSUE TITLE
question about blocking action?

### ISSUE THREAD

**Issue by zhuxiujia** [tokio-rs/tokio#2417]
If I want to run a blocking action, do I need to manually differentiate? Can you automatically To deal with?

```rsut
async fn read_to_string(path: impl AsRef<Path>) -> io::Result<String> {
task::spawn_blocking(move || {
 // to do some thing block io
  std::fs::read_to_string(path)
});
}
```
allow replace to Auto deal with
```rsut
async fn read_to_string(path: impl AsRef<Path>) -> io::Result<String> {
  std::fs::read_to_string(path)
}
```
Because there are many libraries many IO operations are blocked for historical reasons（Even rust Std libraries）
If you use manual differentiation, you'll end up with a huge amount of code。

“async_std” this crate can auto（Inspired by golang）, for example doc.
   https://async.rs/blog/stop-worrying-about-blocking-the-new-async-std-runtime/

**Comment by MikailBag** [comment]
async-std's runtime can not detect blocking calls automatically too AFAIK. That pull request from the blog post was closed.

**Comment by Darksonn** [comment]
Hi. You might want to read [this quite recent documentation improvement][1], which should address these questions. The question is also addressed in [this blog post][2], which also explains why async-std's proposed (but ultimately cancelled) blocking-detection is not comparable to what golang does.

So the answer is: You need to continue using `spawn_blocking` (or `block_in_place`) for blocking operations.

I am closing this issue, but feel free to post additional questions here.

[1]: https://github.com/tokio-rs/tokio/pull/2414
[2]: https://tokio.rs/blog/2020-04-preemption/

**Comment by zhuxiujia** [comment]
> async-std's runtime can not detect blocking calls automatically too AFAIK. That pull request from the blog post was closed.
> UPD: PR itself was [async-rs/async-std#631](https://github.com/async-rs/async-std/pull/631).

no , do you see   https://github.com/async-rs/async-std/pull/733

**Comment by zhuxiujia** [comment]
> Hi. You might want to read [this quite recent documentation improvement](https://github.com/tokio-rs/tokio/pull/2414), which should address these questions. The question is also addressed in [this blog post](https://tokio.rs/blog/2020-04-preemption/), which also explains why async-std's proposed blocking-detection is not comparable to what golang does.
> 
> So the answer is: You need to continue using `spawn_blocking` (or `block_in_place`) for blocking operations.
> 
> I am closing this issue, but feel free to post additional questions here.

Hi,  async_std is successful impl this PR.
if  tokio also can do, it will be  cool

**Comment by Darksonn** [comment]
Tokio does not have any plans to implement such a feature due to its detrimental effects on the tail latencies of the system. The blog post I linked explains why in the [“A note about blocking”][1] section.

[1]: https://tokio.rs/blog/2020-04-preemption/#a-note-on-blocking

**Comment by hawkw** [comment]
> > async-std's runtime can not detect blocking calls automatically too AFAIK. That pull request from the blog post was closed.
> > UPD: PR itself was [async-rs/async-std#631](https://github.com/async-rs/async-std/pull/631).
> 
> no , do you see [async-rs/async-std#733](https://github.com/async-rs/async-std/pull/733)
> 
> #631 PR is a successor impl in #733 . and it is merged.
> This has been merged to master, we will be releasing a 1.6.0-alpha soon

My understanding is that the PR you linked (async-rs/async-std#733) does *not* contain the automatic detection of blocking operations, only internal refactors to the async-std runtime that were factored out from async-rs/async-std#631. 
See https://github.com/async-rs/async-std/pull/631#issuecomment-599376318 and https://github.com/async-rs/async-std/pull/733#issue-391558631.
